src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,698 @@
+                        ADP.TraceExceptionAsReturnValue(e);
+                        throw e;
+                    }
+
+                    if (Interop.Kernel32.GetFileType(hFile) != Interop.Kernel32.FileTypes.FILE_TYPE_DISK)
+                    {
+                        hFile.Dispose();
+                        throw ADP.Argument(SR.GetString(SR.SqlFileStream_PathNotValidDiskResource));
+                    }
+
+                    // if the user is opening the SQL FileStream in read/write mode, we assume that they want to scan
+                    // through current data and then append new data to the end, so we need to tell SQL Server to preserve
+                    // the existing file contents.
+                    if (access == System.IO.FileAccess.ReadWrite)
+                    {
+                        uint ioControlCode = Interop.Kernel32.CTL_CODE(FILE_DEVICE_FILE_SYSTEM,
+                            IoControlCodeFunctionCode, (byte)Interop.Kernel32.Method.METHOD_BUFFERED,
+                            (byte)Interop.Kernel32.IoControlCodeAccess.FILE_ANY_ACCESS);
+
+                        if (!Interop.Kernel32.DeviceIoControl(hFile, ioControlCode, IntPtr.Zero, 0, IntPtr.Zero, 0, out uint cbBytesReturned, IntPtr.Zero))
+                        {
+                            System.ComponentModel.Win32Exception e = new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error());
+                            ADP.TraceExceptionAsReturnValue(e);
+                            throw e;
+                        }
+                    }
+
+                    // now that we've successfully opened a handle on the path and verified that it is a file,
+                    // use the SafeFileHandle to initialize our internal System.IO.FileStream instance
+                    System.Diagnostics.Debug.Assert(_m_fs == null);
+                    _m_fs = new System.IO.FileStream(hFile, access, DefaultBufferSize, ((options & System.IO.FileOptions.Asynchronous) != 0));
+            }
+            catch
+            {
+                if (hFile != null && !hFile.IsInvalid)
+                    hFile.Dispose();
+
+                throw;
+            }
+        }
+
+        #region private helper methods
+
+        // This method exists to ensure that the requested path name is unique so that SMB/DNS is prevented
+        // from collapsing a file open request to a file handle opened previously. In the SQL FILESTREAM case,
+        // this would likely be a file open in another transaction, so this mechanism ensures isolation.
+        static private string InitializeNtPath(string path)
+        {
+            // ensure we have validated and normalized the path before
+            AssertPathFormat(path);
+
+            string formatPath = string.Empty;
+            string uniqueId = string.Empty;
+
+            formatPath = System.IO.PathInternal.IsDeviceUNC(path.AsSpan()) ? @"\??" : @"\??\UNC\{0}\{1}";
```

Comment:
nit: there is an implicit conversion to span from string, you shouldn't need .AsSpan(). Additionally you're not building it right if it is a device path already.  If it is `\\.\UNC\Server\Share\SomePath` you need to change `\\.` to `\??`, not trim off `\\`. You also still need to add your uniqueId.